scripts/pylib/twister/twisterlib.py

```diff
@@ -609,6 +642,8 @@ def handle(self):
             self.instance.reason = "Timeout"
             self.add_missing_testscases(harness)
 
+        self._verify_ztest_suite_name(harness.state, harness.detected_suite_names, handler_time)
```

Comment:
I would suggest this is controlled by an option. and such check will make twister coupled tightly with ztest. and better to implement in a plug in way.

Reply:
I decided to introduce this checking as obligatory due to the fact that it do not impact on tests not written with ztest (for example samples) and it increase only the quality assurance of performed tests. For example this feature  could avoid situation described here:
https://github.com/zephyrproject-rtos/zephyr/issues/42396
Moreover it help to fix this problem by detect all affected tests:
https://github.com/zephyrproject-rtos/zephyr/pull/42439#pullrequestreview-877694014

Reply:
this has a requirement suite name must print in test log, which match its defines in the yaml file. if we decided to do so, we need update the ztest and twister document for this constrains.

Reply:
But, I do not perform this checking basing on any name from yaml file. For verification I use only suite names declared in C file in `ztest_test_suite(...)` or `ZTEST_SUITE(...)` functions.

So I think, that my propositions do not demand any update in ztest or twister documentation. This checking concerns only tests written in Ztest test framework, which requires to pass test suite name to every created test suites.